src/main/java/com/coherent/training/api/kapitsa/base/BaseClientObject.java

```diff
@@ -0,0 +1,124 @@
+package com.coherent.training.api.kapitsa.base;
+
+import lombok.SneakyThrows;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+
+import java.util.Map;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.http.HttpHeaders.AUTHORIZATION;
+
+public class BaseClientObject {
+    private final CloseableHttpClient client;
+    private final HttpUriRequest GETRequest;
+    private final HttpUriRequest POSTRequest;
+    private CloseableHttpResponse response;
+    private final RequestBuilder requestBuilder;
+
+    public BaseClientObject(BaseClientObjectBuilder builder) {
+        this.client = builder.client;
+        this.GETRequest = builder.GETRequest;
+        this.POSTRequest = builder.POSTRequest;
+        this.requestBuilder = builder.requestBuilder;
+    }
+
+    public CloseableHttpClient getClient(){
+        return client;
+    }
+
+    public int getResponseCode(){
+        return response.getStatusLine().getStatusCode();
+    }
+
+    public HttpUriRequest getGETRequest(){
+        return GETRequest;
+    }
+
+    public HttpUriRequest getPOSTRequest(){
+        return POSTRequest;
+    }
```

Comment:
still, why do we need two fields/getters for the request, I should not care about that